format/spec.md

```diff
@@ -298,16 +298,143 @@ Iceberg tables must not use field ids greater than 2147483447 (`Integer.MAX_VALU
+| **`2147483644  _deleted`**       | `boolean`     | Whether the row has been deleted                                                                        |
+| **`2147483643  _spec_id`**       | `int`         | Spec ID used to track the file containing a row                                                         |
+| **`2147483642  _partition`**     | `struct`      | Partition to which a row belongs                                                                        |
+| **`2147483546  file_path`**      | `string`      | Path of a file, used in position-based delete files                                                     |
+| **`2147483545  pos`**            | `long`        | Ordinal position of a row, used in position-based delete files                                          |
+| **`2147483544  row`**            | `struct<...>` | Deleted row values, used in position-based delete files                                                 |
+| **`2147483543  _row_id`**        | `long`        | A unique long assigned when row-lineage is enabled see [Row Lineage](#row-lineage)                      |
+| **`2147483542  _last_update`**   | `long`        | The sequence number which last updated this row when row-lineage is enabled [Row Lineage](#row-lineage) |
+
+### Row Lineage
+
+In Specification V3, an Iceberg Table can declare that engines must track row-lineage of all newly created rows. This
+requirement is controlled by setting the field `row-lineage` to true in the table's metadata. When true, two additional 
+fields in data files will be available for all rows added to the table.
+
+* `_row_id` a unique long for every row. Computed via inheritance for rows in their original datafiles 
+and explicitly written when the row is moved to a new file.
+* `_last_update` the sequence number of the commit which last updated this row. The value is computed via inheritance for 
+rows in their original file or in files where the row was modified.
+
+Commits with `row-lineage` enabled are not allowed to include any [Equality Deletes](#equality-delete-files).
+
+Implementations writing to tables where `row-lineage` is enabled must populate several additional 
+fields in the metadata and propagate row information from existing and updated. 
+
+#### Metadata Propagation
+
+Creating a new commit when `row-lineage` is enabled requires
+* Setting the new snapshot's `first-row-id` field to the previous table metadata's `last-row-id` field
+* Setting `first_row_id` in new data manifest entries in the new manifest list to `first-row-id` field 
+from the snapshot plus the sum of all `added_rows_count` in previously listed new data manifest entries.
+* Setting `first_row_id` in new `data_file` entries  to null and in existing `data_file` structs to their original
+or computed value (if previously null) of `first_row_id`
+* Incrementing the new metadata's `last-row-id` field by the number of new rows added to the table by the commit.
```

Comment:
I think this requirement makes it much harder to implement row lineage because we need to keep track of the number of null values for `_row_id` when writing a data file. Wouldn't it be easier to use `record_count`, which is already a required field?